emmalloc/emmalloc.c

```diff
@@ -545,7 +546,10 @@ static bool claim_more_memory(size_t numBytes)
     // If this is the first time we're called, see if we can use
     // the initial heap memory set up by wasm-ld.
     if (!listOfAllRegions) {
-      unsigned char *heap_end = sbrk(0);
+      unsigned char *heap_end = &__heap_end;
+      if (heap_end == NULL)
+        heap_end = (unsigned char*) ((size_t) &__heap_base + (PAGE_SIZE-1) & -PAGE_SIZE);
```

Comment:
Once non-weak, the `NULL` fallback is not needed, but the dlmalloc code is trapping on `end < base`.